### ISSUE TITLE
[UI Components] Usage of FormGroup's `controlId` resulting duplicate ID 

### ISSUE THREAD

**Issue by huyenltnguyen** [freeCodeCamp/freeCodeCamp#52848]
## Describe the bug

The `FormGroup` component in the `ui-components` library has a `controlId` prop, which helps associate a form control with a label.

We expect that the following code

```jsx
<FormGroup controlId='emailInput'>
  <ControlLabel>Email</ControlLabel>
  <FormControl type='email' />
</FormGroup>
```

would result

```html
<div>
  <label for="emailInput">Email</label>
  <input id="emailInput" type='email' />
</div>
```

However, the result is actually:

```html
<div id="emailInput">
  <label for="emailInput">Email</label>
  <input id="emailInput" type='email' />
</div>
```

This is invalid HTML, and messes up the label-input association, which I think is the cause of #52723.

## Steps to reproduce

1. Go to https://www.freecodecamp.org/update-email/
2. Check the DOM 

<img width="1221" alt="Screenshot 2023-12-31 at 16 00 53" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/25715018/5581d921-ab0d-40fd-a4d0-3eb37e2fb0d7">

Notes: 
- The issue for this particular page will be resolved once #52507 is merged, but it can be reproduced locally with this version of `main`: https://github.com/freeCodeCamp/freeCodeCamp/blob/f8e7ff01dd66b78e202c8488cd79205de20e3cd6/client/src/pages/update-email.tsx. 
- I'm pretty sure we can see the issue in Storybook as well, but I discovered the issue when I was testing the client app.

## Expected behavior

- The id should only be applied to the `input` element.
- Tests need to be added to prevent this issue from happening again.

## Additional context

This issue was discovered in #52507 as we were trying to replace `getByTestId()` with `getByLabel()`. With the `getByLabel` method, the tests failed to find the input as they were confused by the duplicate ID.

## Reference 

Bootstrap FormGroup: https://react-bootstrap-v3.netlify.app/components/forms/

**Comment by dvip1** [comment]
I am new to open source, I want to work on this. I'll send update to this thread if I managed to solve it. Also is that how I'm to ask for a PR ?

**Comment by shootermv** [comment]
i can see the test which checks for existence of id on the `outer` tag
![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/1336862/b7838007-fc88-405d-aae9-d4a890b31d56)

**Comment by altsun** [comment]
I think the root cause of this defect is:
`update-email.tsx` imports `FormGroup` from `@freecodecamp/ui`.
Recently, this PR [#51204](https://github.com/freeCodeCamp/freeCodeCamp/pull/51204/files) updated `tools/ui-components/src/index.ts` to use customized `FormGroup` component.
Customized `FormGroup` component was created in this PR [#46758](https://github.com/freeCodeCamp/freeCodeCamp/pull/46758/files#diff-2fe5d909b80a7c3185b80a57c18033557ed4c28cff2760fc5a690b68b6e868f6R32), I think it assigns `controlId` to `id` of component.

**Comment by Sahillather002** [comment]
do we need to remove the id or change to different one?
